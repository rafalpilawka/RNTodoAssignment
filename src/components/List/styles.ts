import {StyleSheet} from 'react-native';
import normalize from 'react-native-normalize';

export const styles = StyleSheet.create({
  list: {
    flexDirection: 'row',
    height: normalize(80),
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomColor: 'black',
    borderBottomWidth: 1,
  },
  text: {fontSize: normalize(16)},
});
