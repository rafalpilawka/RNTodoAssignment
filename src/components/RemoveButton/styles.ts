import {StyleSheet} from 'react-native';
import normalize from 'react-native-normalize';

export const styles = StyleSheet.create({
  text: {color: 'white', fontSize: 16, fontWeight: 'bold'},
  button: {
    backgroundColor: 'red',
    borderRadius: normalize(50),
    width: normalize(30),
    height: normalize(30),
    justifyContent: 'center',
    alignItems: 'center',
  },
});
