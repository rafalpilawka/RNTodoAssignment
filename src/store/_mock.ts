export const mockTodos = [
  {id: 'aaa', title: 'Add those todos', completed: false},
  {id: 'aaa2233', title: 'Delete second', completed: false},
  {id: 'bbnjhjhjhjh', title: 'Add third one', completed: true},
];

export const mockLists = [
  {name: 'List 1', id: '11242342144'},
  {name: 'list 2', id: '11242342122'},
  {name: 'list 3', id: '1124234211'},
];

export const mockStore = {
  index: mockLists,
  todos: mockTodos,
  loading: false,
  errors: undefined,
  idLoader: undefined,
};

export const mockInitialStore = {
  index: [],
  todos: [],
  loading: false,
  errors: undefined,
  idLoader: undefined,
};
