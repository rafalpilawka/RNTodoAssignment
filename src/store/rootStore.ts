import {all, call} from 'redux-saga/effects';
import {applyMiddleware, combineReducers, createStore, Reducer} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import todosReducer from 'src/store/todos/todos.reducer';
import {watchTodosSaga} from 'src/store/todos/todos.saga';

const sagaMiddleware = createSagaMiddleware();

function* rootSaga() {
  yield all([
    call(watchTodosSaga),
    //OTHER WATCHERS GOES HERE
  ]);
}

// const store: Reducer<RootState> = combineReducers<RootState>({
//   todos: todosReducer,
// });

const rootStore = (state: any, action: any) => {
  return todosReducer(state, action);
};

// export interface RootState {
//   todos: ReduxState;
// }

export type RootState = ReturnType<typeof todosReducer>;

export default createStore(
  rootStore,
  composeWithDevTools(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSaga);
