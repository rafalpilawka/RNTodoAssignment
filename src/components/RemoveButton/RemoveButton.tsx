import React, {ReactElement} from 'react';
import {Pressable, Text} from 'react-native';
import {styles} from 'src/components/RemoveButton/styles';

export type PropsT = {
  onPress(): void;
};

export const RemoveButton = ({onPress}: PropsT): ReactElement => {
  return (
    <Pressable style={styles.button} onPress={onPress}>
      <Text style={styles.text}>x</Text>
    </Pressable>
  );
};
