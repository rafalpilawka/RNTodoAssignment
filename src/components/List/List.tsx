import React, {ReactElement} from 'react';
import {Pressable, Text} from 'react-native';
import {RemoveButton} from 'src/components/RemoveButton/RemoveButton';
import {onShare} from 'src/services/shareList';

import {ListType} from 'src/store/todos/todos.types';
import {styles} from 'src/components/List/styles';

type PropsT = {
  item: ListType;
  navigate(id: string): void;
  removeHandler(id: string): void;
};

export const List = ({
  item: {id, name},
  navigate,
  removeHandler,
}: PropsT): ReactElement => {
  const _removeList = () => {
    removeHandler(id);
  };

  return (
    <Pressable
      style={styles.list}
      onPress={() => navigate(id)}
      onLongPress={() => onShare(id, name)}>
      <Text style={styles.text}>{name}</Text>
      <RemoveButton onPress={_removeList} />
    </Pressable>
  );
};
