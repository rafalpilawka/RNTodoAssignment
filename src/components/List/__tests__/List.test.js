import React from 'react';
import 'react-native';
import {shallow} from 'enzyme';
import { List } from 'src/components/List/List';
import renderer from 'react-test-renderer';
import { mockLists} from 'src/store/_mock';
import {RemoveButton} from 'src/components/RemoveButton/RemoveButton';

const props = {
  item: mockLists[1],
  navigate: jest.fn(),
  removeHandler: jest.fn(),
};

describe('List item', () => {
  let component;
  beforeEach(() => {
    component = shallow(<List {...props} />);
    jest.clearAllMocks();
  });
  it('should match to snapshot', () => {
    const component = renderer.create(<List {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should receive props', () => {
    const nameLabel = component.find({children: 'list 2'});
    expect(nameLabel).toHaveLength(1);
  });

  it('should call removeHandler', () => {
    const removeButton = component.find(RemoveButton);
    removeButton.simulate('press');
    expect(props.removeHandler).toHaveBeenCalledTimes(1);
  });
});
