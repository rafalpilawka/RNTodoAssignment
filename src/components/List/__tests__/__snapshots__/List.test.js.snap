// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List item should match to snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 160,
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
  >
    list 2
  </Text>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 100,
        "height": 60,
        "justifyContent": "center",
        "width": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      x
    </Text>
  </View>
</View>
`;
